From 288c9de564eafcb190afc7677815967587f2d8ca Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@ew.tq-group.com>
Date: Thu, 24 Oct 2024 16:55:34 +0200
Subject: [PATCH] Add generic wrapper for mkimage_imx8

Add a wrapper that tees the stderr output to a log file, allowing for
more robust parsing for CSF / SRK / HAB block offsets.

To keep diff small, compile the real tool as 'mkimage_imx8.real'.
Optimize the clean target: remove $(MKIMG) clean from SoC clean
since the tool is built and called from top level dir.

Add cleaning of the generated log file to the SoC clean target
to prevent using wrong data in subsequent runs.

Upstream-Status: Inappropriate
Signed-off-by: Markus Niebel <Markus.Niebel@ew.tq-group.com>
---
 Makefile        | 14 +++++++-------
 iMX8DXL/soc.mak |  3 ++-
 iMX8QM/soc.mak  |  1 +
 iMX8QX/soc.mak  |  3 ++-
 iMX8ULP/soc.mak |  3 ++-
 iMX91/soc.mak   |  3 ++-
 iMX93/soc.mak   |  3 ++-
 iMX94/soc.mak   |  3 ++-
 iMX95/soc.mak   |  3 ++-
 mkimage_imx8    |  7 +++++++
 10 files changed, 29 insertions(+), 14 deletions(-)
 create mode 100755 mkimage_imx8

diff --git a/Makefile b/Makefile
index b14f9ce..7b39f8a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 
-MKIMG = $(PWD)/mkimage_imx8
+MKIMG_REAL = $(PWD)/mkimage_imx8.real
 CC = gcc
 CFLAGS ?= -g -O2 -Wall -std=c99 -static
 INCLUDE += $(CURR_DIR)/src
@@ -30,10 +30,10 @@ else
 endif
 
 #print out usage as the default target
-all: $(MKIMG) help
+all: $(MKIMG_REAL) help
 
 clean:
-	@rm -f $(MKIMG)
+	@rm -f $(MKIMG_REAL)
 	@rm -f src/build_info.h
 ifdef O
 	@$(MAKE) --no-print-directory -C $(O) -f ../$(SOC_DIR)/soc.mak clean
@@ -48,11 +48,11 @@ endif
 	@$(MAKE) --no-print-directory -C iMX95 -f soc.mak  clean
 	@$(MAKE) --no-print-directory -C iMX94 -f soc.mak  clean
 
-$(MKIMG): src/build_info.h $(SRCS)
-	@echo "Compiling mkimage_imx8"
-	$(CC) $(CFLAGS) $(SRCS) -o $(MKIMG) -I src
+$(MKIMG_REAL): src/build_info.h $(SRCS)
+	@echo "Compiling $@"
+	$(CC) $(CFLAGS) $(SRCS) -o $@ -I src
 
-bin: $(MKIMG)
+bin: $(MKIMG_REAL)
 
 src/build_info.h:
 	@echo -n '#define MKIMAGE_COMMIT 0x' > src/build_info.h
diff --git a/iMX8DXL/soc.mak b/iMX8DXL/soc.mak
index 6dbfc73..79caa0c 100755
--- a/iMX8DXL/soc.mak
+++ b/iMX8DXL/soc.mak
@@ -62,8 +62,9 @@ Image1: Image
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img Image0 Image1 u-boot-hash.bin u-boot-atf.bin head.hash u-boot-atf-container.img flash.bin
+	@rm -f u-boot-atf-container.img Image0 Image1 u-boot-hash.bin u-boot-atf.bin head.hash u-boot-atf-container.img flash.bin
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx8dxl clean done"
 
 flash: $(MKIMG) $(SECO_FW_NAME) scfw_tcm.bin u-boot-atf.bin
diff --git a/iMX8QM/soc.mak b/iMX8QM/soc.mak
index e7d19b0..a6fc8c4 100755
--- a/iMX8QM/soc.mak
+++ b/iMX8QM/soc.mak
@@ -78,6 +78,7 @@ u-boot-atf-container-a72.img: u-boot-atf-a72.bin tee-a72.bin
 clean:
 	@rm -f $(DCD_CFG) .imx8_dcd.cfg.cfgtmp.d $(DCD_800_CFG) $(DCD_1200_CFG) .imx8qm_dcd_800.cfg.cfgtmp.d .imx8qm_dcd.cfg.cfgtmp.d .imx8qm_dcd_1200.cfg.cfgtmp.d head.hash u-boot-hash.bin u-boot-atf-container.img u-boot-atf-hdmi.bin hdmitxfw-pad.bin hdmirxfw-pad.bin flash.bin u-boot-atf.bin u-boot-atf-a72.bin u-boot-hash-a72.bin u-boot-atf-container-a72.img
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx8qm clean done"
 
 flash: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin
diff --git a/iMX8QX/soc.mak b/iMX8QX/soc.mak
index 41791a7..b3d9283 100755
--- a/iMX8QX/soc.mak
+++ b/iMX8QX/soc.mak
@@ -63,8 +63,9 @@ Image1: Image
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) $(DCD_CFG) .imx8qx_dcd.cfg.cfgtmp.d u-boot-atf-container.img Image0 Image1
+	@rm -f $(DCD_CFG) .imx8qx_dcd.cfg.cfgtmp.d u-boot-atf-container.img Image0 Image1
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx8qx clean done"
 
 flash: $(MKIMG) $(AHAB_IMG) scfw_tcm.bin u-boot-atf.bin
diff --git a/iMX8ULP/soc.mak b/iMX8ULP/soc.mak
index 1460bfe..4cd4b2b 100644
--- a/iMX8ULP/soc.mak
+++ b/iMX8ULP/soc.mak
@@ -58,8 +58,9 @@ u-boot-atf-container.img: bl31.bin u-boot-hash.bin
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img
+	@rm -f u-boot-atf-container.img
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx8ulp clean done"
 
 flash_dualboot: $(MKIMG) u-boot-spl.bin u-boot-atf-container.img
diff --git a/iMX91/soc.mak b/iMX91/soc.mak
index 2f975ff..b79b6fd 100644
--- a/iMX91/soc.mak
+++ b/iMX91/soc.mak
@@ -149,8 +149,9 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
+	@rm -f u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx91 clean done"
 
 # Add for System ready
diff --git a/iMX93/soc.mak b/iMX93/soc.mak
index fd522e6..0e63bd4 100644
--- a/iMX93/soc.mak
+++ b/iMX93/soc.mak
@@ -161,8 +161,9 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
+	@rm -f u-boot-atf-container.img u-boot-spl-ddr.bin u-boot-spl-ddr-qb.bin u-boot-hash.bin
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx93 clean done"
 
 # Add for System ready
diff --git a/iMX94/soc.mak b/iMX94/soc.mak
index faaad03..750a75e 100644
--- a/iMX94/soc.mak
+++ b/iMX94/soc.mak
@@ -225,8 +225,9 @@ fcb.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
+	@rm -f u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx94 clean done"
 
 flash_lpboot: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(OEI_IMG_M33)
diff --git a/iMX95/soc.mak b/iMX95/soc.mak
index 501c537..6c0b321 100644
--- a/iMX95/soc.mak
+++ b/iMX95/soc.mak
@@ -294,8 +294,9 @@ flash_fw.bin: FORCE
 
 .PHONY: clean nightly
 clean:
-	@rm -f $(MKIMG) u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
+	@rm -f u-boot-atf-container.img u-boot-spl-ddr-v2.bin m33-oei-ddrfw.bin a55-oei-ddrfw.bin u-boot-hash.bin flash.bin head.hash boot-spl-container.img
 	@rm -rf extracted_imgs
+	rm -f mkimage.log
 	@echo "imx95 clean done"
 
 flash_lpboot: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(OEI_IMG_M33)
diff --git a/mkimage_imx8 b/mkimage_imx8
new file mode 100755
index 0000000..834aba9
--- /dev/null
+++ b/mkimage_imx8
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+OUTLOG=mkimage.log
+SCRIPTPATH=$(dirname "$(readlink -f "${0}")")
+
+
+exec "${SCRIPTPATH}/mkimage_imx8.real" "$@" 2>&1 | tee "${OUTLOG}" >&2
